### PR TITLE
Backend feature/edit delete product

### LINKED ISSUE #231: Implement edit and delete product endpoints

### PR DESCRIPTION
**Pull request by onurkilic2301**
# Description

Implementation of edit and delete product endpoints.

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Unit tests are implemented to test the new endpoints and the code passes all the existing and new tests.

**Test Configuration**:
* OS: MacOS
* Python: 3.8.5

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by onurkilic2301** [comment]
Added endpoints to add photos to specific products and deleting photos of the product.

File: backend/product/views.py [hunk 1]
```diff
@@ -112,3 +113,183 @@ def add_product(request):
...
+    except KeyError:
+        return HttpResponse("Product id not specified", status=400)
+    try:
+        product = Product.objects.get(id=product_id)
+    except Exception:
+        return HttpResponse("There is no such product with the id", status=400)
+    if(vendor != product.vendor):
+        return HttpResponse("You cannot delete products of other vendors", status=401)
+    else:
+        product.delete()
```
Comment (lighterm) [inline backend/product/views.py]: I think this endpoint doesn't delete the image files. It updates the table though.

**Review by lighterm — CHANGES_REQUESTED** [review]
Can you delete the image files also in the delete product endpont.

**Comment by onurkilic2301** [comment]
Requested changes are done!

**Review by lighterm — APPROVED** [review]
Looks great! We can merge it.

**Pull request merged**